shared/src/main/scala/weaponregex/run/TreeMutator.scala

```diff
@@ -7,6 +7,8 @@ import weaponregex.mutator.BuiltinMutators
 object TreeMutator {
   implicit class RegexTreeMutator(tree: RegexTree) {
 
+    // Can these not also be default variables?
+
```

Comment:
It can, and it would look like this
```scala
def mutate(mutators: Seq[TokenMutator] = BuiltinMutators.all, mutationLevel: Int = 0): Seq[Mutant] = {
  val mutatorsFiltered =
    if (mutationLevel == 0) mutators
    else mutators.filter(_.levels.contains(mutationLevel))

  val rootMutants = mutatorsFiltered flatMap (mutator =>
    mutator(tree) map (mutatedPattern =>
      Mutant(mutatedPattern, MutationData(mutator.name, tree.location, mutator.levels.min))
    )
  )

  val childrenMutants = tree.children flatMap (child =>
    child.mutate(mutatorsFiltered) map { case mutant @ Mutant(mutatedPattern, _) =>
      mutant.copy(pattern = tree.buildWith(child, mutatedPattern))
    }
  )

  rootMutants ++ childrenMutants
}
```
Is it better to have default arguments rather than overloading methods?

Reply:
Why is the default mutator level 0? Shouldn't it correspond to whatever level has all the built-in mutators? (2?)

Reply:
A higher-numbered mutation level does not contain all token mutators from the lower-numbered one (e.g. level 
1 is not a subset of level 2). Therefore, if you want to have all the built-in mutators, you cannot pass an actual mutation level number, hence 0 (which will just not filter the mutators). If this is not clear in an API standpoint, mutation level can be made `Option`al, but I don't know how it will be translated into JS.